### PR TITLE
Feature/gather

### PR DESCRIPTION
**Pull request by Spacelocust**
## Description

Please include a summary of the changes and the related issue. Please also include relevant motivation and context. List any dependencies that are required for this change.

Closes #(issue) | No issue linked.

## Type of change

Please add any relevant information to the type of change you've made.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
-   [ ] This change requires a documentation update (or is a documentation update)
-   [ ] HIGH IMPORTANCE (requires a fast merging)

## Has This Been Tested?

Please describe the tests that you ran to verify your changes. It is way more likely for a PR to be merged and approved with successful tests.

-   [x] Yes, I ran tests
-   [ ] No, I did not include any test
-   [ ] No test is needed for this change

### CODE REVIEW

File: api/db/datatype/enum_array.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+		}
+		*a = b
+	}
+
+	return nil
+}
+
+// func (a *EnumArray[T]) GetValues() []T {
+// 	return *a
+// }
```
Comment (YummYume) [inline api/db/datatype/enum_array.go]:
```suggestion
```

File: compose.yml [hunk 2]
```diff
@@ -5,7 +5,8 @@ services:
       dockerfile: ./docker/dev/golang/Dockerfile
     restart: unless-stopped
     container_name: fd-api
-    command: wgo run main.go
+    command: wgo run main.go server
+    # command: sh -c "tail -f /dev/null"
```
Comment (YummYume) [inline compose.yml]:
```suggestion
```

File: compose.yml [hunk 3]
```diff
@@ -18,6 +19,8 @@ services:
       - 5000:5000
     volumes:
       - ./api:/usr/src/app
+    depends_on:
+      - postgres
```
Comment (YummYume) [inline compose.yml:22]: Healthcheck ?

**Review by YummYume — APPROVED** [review]

File: compose.yml (hunk 3 above)
Comment (Spacelocust) [reply compose.yml:22]: Le healthcheck est présent pour le postgres

**Review by YummYume — APPROVED** [review]

**Pull request merged**